source/kraken/apply_disruption.cpp

```diff
@@ -442,32 +470,18 @@ struct add_impacts_visitor : public apply_impacts_visitor {
 
             new_vp.days = new_vp.days & (vj->validity_patterns[rt_level]->days >> vj->shift);
 
-            auto* new_vj = mvj->create_discrete_vj(new_vj_uri,
+
+            auto* new_vj = create_vj_from_old_vj(
+                    mvj,
+                    vj,
+                    new_vj_uri,
                     rt_level,
                     new_vp,
-                    vj->route,
                     std::move(new_stop_times),
                     pt_data);
+            vj = nullptr; //after the call to create_vj, vj may have been deleted :(
```

Comment:
But, it is already done here ?
https://github.com/CanalTP/navitia/pull/2613/commits/312b1d6da9bba5f6e197247022ff8fcee326fa73#diff-c234f9d9583465696c5f3311313759d0R75

Reply:
That's local variables